File: Spring-jpa-criteria/src/main/java/nel/marco/db/entity/Customer.java
package nel.marco.db.entity;

import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.Date;

@Entity
@Table
@Data
public class Customer {

  @Id
  @GeneratedValue(strategy = GenerationType.SEQUENCE)
  private long id;

  @Column private String name;
  @Column private int age;

  @Column
  @Temporal(TemporalType.TIMESTAMP)
  @CreationTimestamp
  private Date created;

  @Column
  @UpdateTimestamp
  @Temporal(TemporalType.TIMESTAMP)
  private Date updated;

  @Enumerated(EnumType.STRING)
  private ActiveType activeType;

  public enum ActiveType {
    ACTIVE,
    DEACTIVATED,
    SUSPENDED
  }
}

File: scratchpad/postgress-returning-lockout/src/main/kotlin/nel/marco/db/Customer.kt
package nel.marco.db

import jakarta.persistence.*
import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.GenericGenerator
import org.hibernate.annotations.UpdateTimestamp
import java.time.OffsetDateTime
import java.util.*


@Entity
@Table
data class Customer(
    @Id
    @GeneratedValue(generator = "uuid2")
    var id: UUID? = null,

    @Column
    var name: String? = null,

    @CreationTimestamp
    var created: OffsetDateTime = OffsetDateTime.now(),

    @UpdateTimestamp
    var updated: OffsetDateTime = OffsetDateTime.now(),
)